axum/src/serve.rs

```diff
@@ -2,24 +2,29 @@
 
 use std::{
     convert::Infallible,
+    fmt::Debug,
     future::{Future, IntoFuture},
     io,
     marker::PhantomData,
     net::SocketAddr,
     pin::Pin,
+    sync::Arc,
     task::{Context, Poll},
     time::Duration,
 };
 
 use axum_core::{body::Body, extract::Request, response::Response};
-use futures_util::future::poll_fn;
+use futures_util::{future::poll_fn, pin_mut, FutureExt};
```

Comment:
While `std::pin::pin` is unfortunately not available with our MSRV, `std::future::poll_fn` is. Might make sense to use it so it becomes more apparent where we *actually* depend on the util crate.